.github/workflows/docfx-gh-pages.yml

```diff
@@ -0,0 +1,51 @@
+name: Deploy DocFX with GitHub Pages dependencies preinstalled
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["feature/docfx"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Github Pages
+        uses: actions/configure-pages@v3
+        
+      - name: Build DocFX Site
+        uses: VakuWare/docfx-pdf-action@v1.4.0
+        with:
+         args: docs/v2/docfx.json
+      - name: Upload docfx built site artifact
+        uses: actions/upload-pages-artifact@v1
+        with:
+         path: docs/v2/_site
```

Comment:
There is a name parameter here which defaults to github-pages (see https://github.com/actions/upload-pages-artifact/blob/main/action.yml), then below you are targeting the environment 'github-pages'. I'm wondering if that is how this is all related. 

One guess to get the docs under a v2 published directory is to change how the docfx json is configured so that it publishes to _site/v2 instead of just _site. Then, when the deployment happens, it deploys from _site, which will have the docfx deployed _site/v2 folder in it?

Reply:
I gave that a go then reverted. It did look like it tried to deploy. Not sure why it 404d

Reply:
OK, i'll get this merged in a play around with it.